system/integrations/checkmk/api-reference.rst

```diff
@@ -0,0 +1,173 @@
+
+   If you use another monitoring tool that’s not officially supported by Zammad,
+   there’s probably a way to make it work with your Checkmk callback URL.
+
+Checkmk Parameters
+------------------
+
+When a notification is received, Zammad creates a new article
+containing the details of the event that triggered it:
+
+.. figure:: /images/system/integrations/checkmk/checkmk-parameters.png
+   :alt: Checkmk article body
+   :align: center
+
+These details come from the fields listed below,
+which correspond to parameters provided by Checkmk (``$NOTIFY_*``).
+Technically, they can be replaced with whatever you want,
+but it’s not recommended.
+
+**Required fields are marked with an asterisk (\*).**
+
+event_id\*
+   A unique ID for the system event. (``$NOTIFY_SERVICEPROBLEMID`` / ``$NOTIFY_HOSTPROBLEMID``)
+
+host\*
+   The hostname of the system that the event originated from. (``$NOTIFY_HOSTNAME``)
+
+   Used to determine if a new event belongs to an existing ticket.
+   Also used in the subject line of the resulting article (“<host> is <state>”).
+
+service
+   The name of the service that the event originated from. (``$NOTIFY_SERVICEDESC``)
+
+   Used to determine if a new event belongs to an existing ticket.
+
+   Displayed as ``-`` when omitted.
+
+state\*
+   The current state of the service or host in question. (``$NOTIFY_SERVICESTATE`` / ``$NOTIFY_HOSTSTATE``)
+
+   Used to detect when a ticket should be auto-closed (*i.e.,* on ``OK``/``UP``).
+   Also used in the subject line of the resulting article (“<host> is <state>”).
+
+text
+   The output of the process that triggered the event. (``$NOTIFY_SERVICEOUTPUT`` / ``$NOTIFY_HOSTOUTPUT``)
+
+   Displayed as ``-`` when omitted.
+
+Ticket Attributes
+-----------------
+
+.. figure:: /images/system/integrations/checkmk/finding-object-names.png
+   :alt: The Object Manager attribute panel displays built-in and custom attribute names.
+   :align: center
+   :width: 80%
+
+   Find a complete list of ticket attributes in the Object Manager.
+
+Pass ticket attributes to change the settings of the ticket
+that will be created for this event.
+(Note that these attributes will be ignored
+if the event belongs to an existing ticket.)
+
+Why would you want to do this?
+Maybe you have only one IT guy,
+and all system monitoring issues should be automatically assigned to him.
+Or, maybe you’re creating multiple notification rules
+so that database outages take higher priority
+than disk space warnings.
+
+In most cases, you’ll probably want to set one of the following:
+
+* group
+* owner
+* state
+* priority
+
+but in practice, you can set almost any attribute,
+including :doc:`custom ones you created through the Object Manager </system/objects>`.
+
+.. note:: 🙅 The following attributes are **not customizable**:
+
+   * title
+   * id
+   * ticket number
+   * customer
+   * created_by_id
+   * updated_by_id
+
+How do I know what values I can set?
```

Comment:
While I love the rewrite of the above, I noticed the part below misses information from my original commit.
This especially hits the fan in terms of id based attributes.

e.g. the possibility to use either `group` or `group_id` if you like.
Depending on the flavor you prefer.

Do you think this is too much input and "named" attributed are good enough?

Reply:
Ah, you're right—I forgot to mention that I removed this.

I will prepare a rewrite that covers this point, but IMO this subject really deserves its own article. Understanding how to find the `group_id` for a given group requires and understanding of DB relations and the underlying structure of Zammad's data model, which is something that I wouldn't expect the average reader to know about or be able to figure out on their own. Ideally, this article could live in the "Danger zone (for experts)" section of the zammad-documentation repo, and we could link there from this page.

In reality, I don't expect that to happen anytime soon, so I'll just make a small note of it here.

Reply:
This is addressed in the latest commit. Leaving this conversation open so that you can a) resolve it if you're okay with it, or b) continue the discussion if you're not.